### DISCUSSION TITLE
Dynamic url param in useForm['Create'] resource

### DISCUSSION THREAD

**Discussion by w20k** [refinedev/refine#6328] [Q&A] [answered]
Hey Refine Team,

I wanted to bring up an interesting topic for discussion. Right now, I have an ugly solution (haha). It works, but I'd like to know if there’s a better way to achieve it within the framework's boundaries.

Here’s the context: I have a create form where a post is connected as a child through a URL parameter: /parents/${parent_id}/childs. I need to update this URL based on the selected ID from the form.

Here’s what I’m currently doing:
```
const [selectedId, setSelectedId] = useState<string>('')

const { formProps, saveButtonProps } = useForm({
    resource: `parents/${selectedFarmId}/childs`,
    action: 'create',
    redirect: false,
})

...................................
<Form.Item
          label={'Parent Name'}
          name="parentId"
          rules={[{ required: true }]}
        >
          <Select
            {...parentSelectProps}
            onChange={value => setSelectedId(value as unknown as string)}
          />
</Form.Item>

```

Do you think there's a more efficient way to handle this? I do think there should be an option to change url before send, but haven't found yet elegant way.

Thanks!

**Comment by alicanerdurmaz** [comment, marked as answer]
Hello @w20k, 

Yes, it doesn't look great, but it's correct for Refine's current state. We're planning improvements in Refine@5, including custom actions and type support for nested paths.

Thank you very much for your use case!

**Reply by w20k** [reply]
Hello @alicanerdurmaz,

Thank you for your answer!

**Reply by w20k** [reply]
Hi @alicanerdurmaz,

It won't work, found a bug with antd. Not yet sure where exactly, but it resets the form to initial state, most probably because resource URL is changing. You can sort of hack through by pressing multiple times, but as it seems easier to use custom mutation to update.
